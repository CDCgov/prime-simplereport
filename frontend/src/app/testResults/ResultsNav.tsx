import React from "react";

import { LinkWithQuery } from "../commonComponents/LinkWithQuery";

const ResultsNav = () => {
  const classNameByActive = ({ isActive }: { isActive: boolean }) =>
    isActive ? "active" : "";

  return (
    <nav className="prime-secondary-nav" aria-label="Secondary navigation">
      <ul className="usa-nav__secondary-links prime-nav">
        <li className="usa-nav__secondary-item">
          <LinkWithQuery to={`/results/1`} end className={classNameByActive}>
            View test results
          </LinkWithQuery>
        </li>
        <li className="usa-nav__secondary-item">
          <LinkWithQuery
            to={`/results/upload`}
            end
            className={classNameByActive}
          >
            Upload spreadsheet
          </LinkWithQuery>
        </li>
        <li className="usa-nav__secondary-item">
          <LinkWithQuery
            to={`/results/upload/submissions`}
            className={classNameByActive}
          >
            View upload history
          </LinkWithQuery>
        </li>
      </ul>
    </nav>
  );
};

export default ResultsNav;
