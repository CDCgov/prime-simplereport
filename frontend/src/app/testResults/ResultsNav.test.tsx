import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import ResultsNav from "./ResultsNav";

describe("ResultsNav", () => {
  it("displays the nav order correctly and defaults to 'Test Results'", () => {
    const { container } = render(
      <MemoryRouter>
        <ResultsNav />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
