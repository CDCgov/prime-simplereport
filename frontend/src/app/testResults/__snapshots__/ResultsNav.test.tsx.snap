// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsNav displays the nav order correctly and defaults to 'Test Results' 1`] = `
<div>
  <nav
    aria-label="Secondary navigation"
    class="prime-secondary-nav"
  >
    <ul
      class="usa-nav__secondary-links prime-nav"
    >
      <li
        class="usa-nav__secondary-item"
      >
        <a
          class=""
          href="/results/1"
        >
          View test results
        </a>
      </li>
      <li
        class="usa-nav__secondary-item"
      >
        <a
          class=""
          href="/results/upload/submit"
        >
          Upload spreadsheet
        </a>
      </li>
      <li
        class="usa-nav__secondary-item"
      >
        <a
          class=""
          href="/results/upload/submissions"
        >
          View upload history
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
